src/DynamoCore/Models/DynamoModelEvents.cs

```diff
@@ -90,15 +90,16 @@ public virtual void OnRequestLayoutUpdate(object sender, EventArgs e)
         /// <summary>
         /// Occurs before current workspace is cleared
         /// </summary>
-        public event Action WorkspaceClearing;
+        public event Action<WorkspaceModel> WorkspaceClearing;
 
         /// <summary>
         /// Triggers WorkspaceClearing event
         /// </summary>
-        public virtual void OnWorkspaceClearing()
+        /// <param name="workspace">Workspace about to be cleared</param>
+        public virtual void OnWorkspaceClearing(WorkspaceModel workspace)
```

Comment:
This is an API change. I believe this is ok as long as we document it [here](https://github.com/DynamoDS/Dynamo/wiki/API-Changes-in-Dynamo-2.12) ?

Reply:
That is incorrect. The public API should not be changed unless the major version number is incremented. Instead obsolete the old method and add a new one if possible.